File: Entities/Employee.cs
using System;
using System.Collections.Generic;

namespace Entities
{
    public class Employee
    {
        public string SpouseName { get; set; }
        public IList<string> Children => new List<string>();
        public int Id { get; set; }
        public string Name { get; set; }

        public Employee()
        {
        }

        public Employee(int id, string name)
        {
            this.Id = id;
            this.Name = name;
        }

    }
}

File: EmployeeCostCalculator/ClientApp/src/app/models/employee.ts
import { CostConfiguration } from '../services/cost-configuration';

export class Employee {
  public spouseName: string;
  public children: string[] =[];

  //TODO: create view model and move those properties to it
  public cost: number;
  public get balance(): number{
    if(isNaN(this.cost)) return null;
    return CostConfiguration.paycheckSalary - this.cost;
  }
  constructor(
    public id: number, // id should be immutable
    public name: string,
  ) {
  }

  public get childrenNames() {
    if (!this.children || this.children.length === 0) {
      return '';
    }
    return this.children.toString();
  }

}
